### PR TITLE
master lpd 31137

### PR DESCRIPTION
**Pull request by kyle-miho**
https://liferay.atlassian.net/browse/LPD-31137
cc: @nicolas-sss @drewbrokke

### CODE REVIEW

**Comment by kyle-miho** [comment]
ci:test:hotel

**Comment by kyle-miho** [comment]
ci:test:sf

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseFragmentCollectionContributorExtendedClassesCheck.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+		throw new Exception(
+			StringBundler.concat(
+				"Could not find FragmentCollectionKey in Java Class ",
+				javaClass.getName(), ". Please add FragmentCollectionKey to ",
+				"method getFragmentCollectionKey"));
+	}
+
+	private static final Pattern _pattern = Pattern.compile(
+		"return \"(.*?)\";");
```
Comment (drewbrokke) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseFragmentCollectionContributorExtendedClassesCheck.java:77]: Are there any cases where a constant is used?

**Review by drewbrokke — COMMENTED** [review]
Just the one question, otherwise this looks good.

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseFragmentCollectionContributorExtendedClassesCheck.java (hunk 1 above)
Comment (kyle-miho) [reply modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseFragmentCollectionContributorExtendedClassesCheck.java:77]: @drewbrokke no, they are all hardcoded to be a string to be of the format ([A-Z]+_?)+ such as TEST_PATTERN
They are not supposed to have any logic in them either

**Comment by kyle-miho** [comment]
@drewbrokke I did a double check and I forgot one thing, can you double check it? This should be fine since its used in other files too, I also re-tested it and it still passed testing and SF

**Review by drewbrokke — APPROVED** [review]

**Comment by kyle-miho** [comment]
@nicolas-sss I assume this pull is good to go though, so im going to push it through. In this case, even if we avoid inheritance checking, it requires java code since it needs to grab a return value. I can see the value for that, Ill create a spike to investigate the value of adding inheritence checking for UpgradeCatchAllCheck

**Comment by kyle-miho** [comment]
https://liferay.atlassian.net/browse/LPD-31838 fyi

**Pull request closed without merging**